clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -48,6 +57,17 @@ class ElasticsearchIndex[CI <: ClioIndex](
     s"$rootDir$dir/"
   }
 
+  /**
+    * The ID of a json record in elasticsearch. By default, each record's ID consists
+    * of the concatenated values of the Key fields.
+    * Additional fields can be added to the ID using the ElasticsearchIndex constructor.
+    */
+  def getId(json: Json): String = {
+    val keyFields = FieldMapper[clioIndex.KeyType].fields.keys
+      .map(snakeCaseTransformation)
+    (keyFields ++ additionalIdFields).map(json.unsafeGet[String]).mkString
```

Comment:
I'm not sure there's a generic way to decode this... you might need to go through circe's api for working with JSON and handle the cases for string and number explicitly (and I guess throw for all the other cases).